### ISSUE TITLE
Ability yield to an Execution

### ISSUE THREAD

**Issue by cowboyd** [thefrontside/effection.js#28]
It's a commonly emerging pattern to be able fork off an operation that itself has several parallel forks within it. Then, at some later point, you want to wait until all the children are finished. An example of this is the [piping that happens in the bigtest proxy server][1]. There are two or three async operations copying data through the pipe, but once they're all finished only then do we want to continue.

In order to support this we need a way to yield to an execution. E.g.

```js
yield fork(function*() {
  fork(one);
  fork(two);
  yield on(emitter, "end");
}
```

In other words, we want to wait until all sub children are finished.

[1]: https://github.com/bigtestjs/server/pull/10/files#r343671254

**Comment by cowboyd** [comment]
As of #57, you can now `join` an execution with the `join` operator:

```ts
let wait = yield fork(timeout(5000));
yield join(wait);
```
